src/sample.jl

```diff
@@ -375,8 +382,18 @@ function mcmcsample(
     nchains::Integer;
     progress = PROGRESS[],
     progressname = "Sampling ($(Distributed.nworkers()) processes)",
+    rngs::Union{Nothing, AbstractVector{<:Random.AbstractRNG}} = nothing,
     kwargs...
 )
+    if rngs === nothing
+        rngs = [deepcopy(rng) for _ in 1:nchains]
+        # Create a seed for each chain using the provided random number generator.
+        seeds = rand(rng, UInt, nchains)
+        Random.seed!.(rngs, seeds)
```

Comment:
There is no need to invoke the broadcasting machinery here.
```suggestion
        foreach(Random.seed!, rngs, seeds)
```